### PR TITLE
MGMT-2769: allow bonding interfaces to be returned in the inventory

### PR DESCRIPTION
**Pull request by asalkeld**

### CODE REVIEW

File: src/inventory/interfaces.go [hunk 1]
```diff
@@ -49,12 +50,26 @@ func (n *NetworkInterface) Addrs() ([]net.Addr, error) {
...
 	if err != nil {
-		logrus.WithError(err).Warnf("Could not determin if interface %s is physical", n.netInterface.Name)
+		logrus.WithError(err).Warnf("Could not determine if interface %s is physical", n.netInterface.Name)
 		return true
 	}
 	return !strings.Contains(evaledPath, "/virtual/")
 }
 
+func (n *NetworkInterface) IsBonding() bool {
+	evaledPath, err := n.dependencies.EvalSymlinks(fmt.Sprintf("/sys/class/net/%s/bonding", n.netInterface.Name))
```
Comment (ori-amizur) [inline src/inventory/interfaces.go:60]: Why not use netlink library? It can be used to detect the interface type.  We already use it.

**Comment by ronniel1** [comment]
@asalkeld How do we test this?
Subsystem, e2e test?

File: src/inventory/interfaces.go (hunk 1 above)
Comment (asalkeld) [reply src/inventory/interfaces.go:60]: sure, I can change that

**Comment by asalkeld** [comment]
> @asalkeld How do we test this?
> Subsystem, e2e test?

I am not sure what e2e testing infra we have. I tested by creating a bonding interface on my laptop and running
```
go run ./src/inventory/main/main.go -with-text-logging | jq ".interfaces"
```

**Comment by asalkeld** [comment]
I can't figure out how to retest this from Jenkins UI

**Pull request merged**